sample/weighted.go

```diff
@@ -0,0 +1,98 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this code is governed by a BSD-style
+// license that can be found in the LICENSE file
+
+package sample
+
+import "math/rand"
+
+// Weighted provides sampling without replacement from a collection of items with
+// weighted probabilites.
```

Comment:
It's not wrong, it's just an overloading of the word "weight". It is a weighted sampling, but the weights is proportional, not equal to the weight.

How about, "of items with non-uniform probability"